pkg/reconciler/common/transformers.go

```diff
@@ -30,7 +30,9 @@ func transformers(ctx context.Context, obj v1alpha1.KComponent) []mf.Transformer
 		mf.InjectNamespace(obj.GetNamespace()),
 		ImageTransform(obj.GetSpec().GetRegistry(), logger),
 		ConfigMapTransform(obj.GetSpec().GetConfig(), logger),
-		ResourceRequirementsTransform(obj.GetSpec().GetResources(), logger)}
+		ResourceRequirementsTransform(obj.GetSpec().GetResources(), logger),
+		JobTransform(obj, logger),
```

Comment:
I don't think we need the logger for anything here (probably not in the others either, but we can deal with that later). Whatever error is caught by `stages.Execute` will be logged.